### ISSUE TITLE
[RFE] Expose Kerberos ticket data on IFP from KCM

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1501]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1497
- Created at 2012-08-23 05:12:36 by [nalin](https://pagure.io/user/nalin)
- Assigned to nobody
---

If SSSD registered on the system message bus, then interested clients could listen for signals indicating when a user's Kerberos credentials were about to expire.  Preferably, SSSD could be enlisted to monitor credentials that it didn't acquire, and provide a dialog-based method for obtaining new initial credentials.

This takes per-cctype work for figuring out how to monitor a ccache off of the desktop's plate.  It also solves a problem that an unprivileged user has a harder time doing: if SSSD has access to the system keytab, it can use that to obtain credentials to use for setting up FAST.  In realms where anonymous PKINIT isn't available, that's the only dependable option for doing so, and an unprivileged user can't do it alone.

### Comments

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2012-08-23 16:03:04](https://pagure.io/SSSD/sssd/issue/1497#240077)**

Fields changed

milestone: NEEDS_TRIAGE => Temp milestone
proposed_priority: Undefined => Core
rhbz:  => todo
summary: Add a general-purpose D-Bus responder => [RFE] Add a general-purpose D-Bus responder

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2012-08-29 17:08:20](https://pagure.io/SSSD/sssd/issue/1497#240078)**

Fields changed

summary: [RFE] Add a general-purpose D-Bus responder => [RFE] Add a general-purpose D-Bus responder for ticket monitoring

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2012-09-04 23:20:19](https://pagure.io/SSSD/sssd/issue/1497#240079)**

Moving all the features planned for 1.10 release into 1.10 beta.

milestone: Temp milestone => SSSD 1.10 beta

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2012-09-04 23:46:02](https://pagure.io/SSSD/sssd/issue/1497#240080)**

Fields changed

priority: minor => critical

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2012-12-20 21:44:55](https://pagure.io/SSSD/sssd/issue/1497#240081)**

Fields changed

design:  => 
design_review:  => 0
fedora_test_page:  => 
selected:  => Not need

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2013-01-02 15:32:08](https://pagure.io/SSSD/sssd/issue/1497#240082)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2013-10-24 17:18:16](https://pagure.io/SSSD/sssd/issue/1497#240083)**

If SSSD could be enlisted (via message bus or other means) to monitor credentials that it didn't acquire, it would also help solve a couple of other use-cases on servers where automatic renewal is useful:

    - people logging in via SSH PKI (with keys), running kinit manually, and then running long-running jobs that access e.g. NFS with KRB5
    - people logging in via SSH GSS-API (i.e. passwordless logins) and then running long-running jobs, etc.

Currently tickets won't be renewed automatically in those cases as the credentials cache is created by either kinit or sshd and not SSSD.

If appropriate functionality was exposed, it would be possible to script initialising tickets and adding the cache to be monitored from e.g. login scripts.

Also see related [discusson on sssd-users list](https://lists.fedorahosted.org/pipermail/sssd-users/2013-September/001014.html).

changelog:  => 
review:  => 0

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2013-10-25 20:06:08](https://pagure.io/SSSD/sssd/issue/1497#240084)**

Replying to [comment:7 mighg]:
> If SSSD could be enlisted (via message bus or other means) to monitor credentials that it didn't acquire, it would also help solve a couple of other use-cases on servers where automatic renewal is useful:
> 
>  - people logging in via SSH PKI (with keys), running kinit manually, and then running long-running jobs that access e.g. NFS with KRB5
>  - people logging in via SSH GSS-API (i.e. passwordless logins) and then running long-running jobs, etc.

Please take a look at the GSSProxy project. GSSproxy can auto acquire and renew the tickets on your behalf in the scenarios that you described allowing scripts running as a user on a box assume kerberos identity you find appropriate.

https://fedorahosted.org/gss-proxy/ [[BR]] 
https://fedoraproject.org/wiki/Features/gss-proxy

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2014-02-14 10:15:06](https://pagure.io/SSSD/sssd/issue/1497#240085)**

Yes, gss-proxy does sound interesting indeed, but as Jakub Hrozek noted in the [discussion](https://lists.fedorahosted.org/pipermail/sssd-users/2013-September/001027.html): *due to a large number of dependencies, GSSProxy is probably never coming to RHEL-6*

cc:  => Michael.Gliwinski@henderson-group.com

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2014-09-26 18:49:56](https://pagure.io/SSSD/sssd/issue/1497#240086)**

Fields changed

mark:  => 0

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2015-02-10 13:55:20](https://pagure.io/SSSD/sssd/issue/1497#240087)**

Fields changed

milestone: SSSD 1.13 beta => SSSD 1.13 backlog
patch: 0 => 1
priority: critical => minor

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2015-02-12 20:26:38](https://pagure.io/SSSD/sssd/issue/1497#240088)**

Mass-moving tickets not planned for the next two releases.

Please reply with a comment if you disagree about the move..

milestone: SSSD 1.13 backlog => SSSD 1.15 beta

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2016-11-25 10:40:38](https://pagure.io/SSSD/sssd/issue/1497#240089)**

Fields changed

blockedby:  => #2887
sensitive:  => 0

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2017-02-24 15:01:36](https://pagure.io/SSSD/sssd/issue/1497#240090)**

**Metadata Update from @nalin**:
- Issue marked as depending on: #2887
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2017-03-03 11:57:33](https://pagure.io/SSSD/sssd/issue/1497#429234)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch adjusted to on (was: 1)
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset
- Issue close_status updated to: None

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2017-03-03 11:57:45](https://pagure.io/SSSD/sssd/issue/1497#429236)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2017-11-16 08:55:35](https://pagure.io/SSSD/sssd/issue/1497#479648)**

**Metadata Update from @lslebodn**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue tagged with: KCM

---

**Comment from [nalin](https://pagure.io/user/nalin) at [2020-03-13 14:23:48](https://pagure.io/SSSD/sssd/issue/1497#633191)**

**Metadata Update from @thalman**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue tagged with: Future milestone
